test/cmdlineargs.jl

```diff
@@ -139,7 +139,9 @@ let exename = `$(Base.julia_cmd()) --startup-file=no --color=no`
     # handling of @projectname in --project and JULIA_PROJECT
     let expanded = abspath(Base.load_path_expand("@foo"))
         @test expanded == readchomp(`$exename --project='@foo' -e 'println(Base.active_project())'`)
-        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir()))
+        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir(), "JULIA_DEPOT_PATH" => "$(Base.DEPOT_PATH[1])"))
```

Comment:
I think we should actually just delete the `JULIA_DEPOT_PATH` environment variable.

```suggestion
        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir(), "JULIA_DEPOT_PATH" => nothing))
```

Reply:
Why? That won't work, because it would essentially point it to a folder named nothing in the current working directory. At least that's how it plays out on my machine rn...

Reply:
`"JULIA_DEPOT_PATH" => nothing` is equivalent to doing `unset JULIA_DEPOT_PATH` in Bash or doing `delete!(ENV, "JULIA_DEPOT_PATH") in Julia.

Reply:
Hmmmm. At least, that is what should happen. It doesn't seem to be working correctly.

Reply:
But we want to make sure that that the created env is in wherever the user is setting it... removing it will take us back to square one, if not worse. I am still confused how this will resolve the issue.

Reply:
I guess it works for `withenv`, but not for `addenv` or `setenv`.